### PR TITLE
 Add pricesheet download api to Microsoft.Billing (DoNotMerge)

### PR DESCRIPTION
**Pull request by sandeepkatumalla**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.

### CODE REVIEW

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-09-30/consumption.json [hunk 1]
```diff
@@ -0,0 +1,4538 @@
+{
```
Comment (sandeepkatumalla) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-09-30/consumption.json]: Please ignore. This was already reviewed on a previous PR. Adding here for completeness sake so checks pass.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json [hunk 2]
```diff
@@ -275,6 +275,68 @@
         }
       }
     },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/invoices/{invoiceName}/pricesheet/download": {
```
Comment (sandeepkatumalla) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json]: This is the only new swagger addition that needs to be reviewed.

**Comment by annatisch** [comment]
Hi @sandeepkatumalla - there appears to be an error in the Java generation configuration.
The autorest error is:
"An item with the same key has already been added. Key: invoices"

So it looks like there's a duplicate reference in there somewhere.

**Comment by annatisch** [comment]
Adding @ravbhatnagar for ARM review.

@sandeepkatumalla - there are currently two linter warnings, could you please take a look?
```
The operation 'Invoices_Pricesheet' returns 202 status code, which indicates a long running operation, please enable "x-ms-long-running-operation.
--
'GET' operation 'Invoices_Pricesheet' should use method name 'Get' or Method name start with 'List'. Note: If you have already shipped an SDK on top of this spec, fixing this warning may introduce a breaking change.
```

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json (hunk 2 above)
Comment (ravbhatnagar) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json]: This API is not modeled correctly.  Few things -
1. Name of the pricesheet is missing from this API. if pricesheets is a singleton, you can have the name hardcoded to "default. 
2. how will a GET on pricesheet/default be different from the download action
3. actions are modeled as a POST. So in the scenario, which I would like to understand, that you want to provide the download capability, it should be modeled as a POST .../pricesheets/default/download
4. ARM does not allow download kind of capabilities. If you want to provide a way for customers to download the pricesheet, you can provide a pointer to the storage account blob from where they can download the pricesheet. And if this were the case, you dont need a separate action, downloadURL can be a property on the pricesheet entity itself. 
Hope that makes sense. let me know if you have questions.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json [hunk 3]
```diff
@@ -275,6 +275,68 @@
...
+          {
+            "name": "invoiceName",
+            "description": "The name of an invoice resource.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "202": {
```
Comment (ravbhatnagar) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json]: Final response of this API MUST be modeled. what happens when this completes?

**Review by ravbhatnagar — COMMENTED** [review]
@sandeepkatumalla please take a look at the comments.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json (hunk 2 above)
Comment (sandeepkatumalla) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json]: Updated review as per our offline conversation. Please verify.

**Comment by ravbhatnagar** [comment]
Looks good. Thanks for making the changes

**Comment by annatisch** [comment]
Thanks @sandeepkatumalla - just one model validation error left. Could you please add the 200 response to your example?
```
error : 
operationId: InvoicePricesheets_Download
scenario: pricesheetsDownload
source: response
responseCode: '200'
errorCode: RESPONSE_STATUS_CODE_NOT_IN_EXAMPLE
errorDetails:
  message: >-
    Following response status codes "200" for operation
    "InvoicePricesheets_Download" were present in the swagger spec, however they
    were not present in x-ms-examples. Please provide them.
```

**Comment by annatisch** [comment]
Thanks @sandeepkatumalla - though there's still an error in the example:
```
 error : 
operationId: InvoicePricesheet_Download
scenario: pricesheetDownload
source: response
responseCode: '200'
errorDetails:
  code: OBJECT_ADDITIONAL_PROPERTIES
  params:
    - - value
  message: 'Additional properties not allowed: value'
  path: ''
  title: /definitions/DownloadUrl
  description: A secure URL that can be used to download a an entity until the URL expires.
```
The 200 response body is not intended to be a list is it?

**Comment by annatisch** [comment]
Review complete!

**Pull request closed without merging**